### PR TITLE
Initial Stripe Integration

### PR DESCRIPTION
**Pull request by djoyal**
This PR adds a Credit Card input (provided by Stripe) to the `/checkout` page. 

This input is currently enabled because a fake `client_secret` has been provided to the component. This `client_secret` should actually be generated by the backend during runtime and returned to the frontend for usage. However this requires a rework of the backend code to enable, as the original workflow (planned for the mobile app) is not applicable to the web workflow.

Screen Shot 2021-02-11 at 11 14 23 AM

### CODE REVIEW

File: packages/web/.env.example [hunk 1]
```diff
@@ -7,4 +7,6 @@ NEXT_PUBLIC_AUTH0_AUDIENCE=
 
 NEXT_PUBLIC_API_HOST=https://pbdl-api-dev.azurewebsites.net
 NEXT_PUBLIC_API_PREFIX=api/v1
-NEXT_PUBLIC_MEDIA_HOST=https://rgbkdiag.blob.core.windows.net/blob-pbdl-media-dev
\ No newline at end of file
+NEXT_PUBLIC_MEDIA_HOST=https://rgbkdiag.blob.core.windows.net/blob-pbdl-media-dev
+
+STRIPE_KEY=
```
Comment (jfplataroti) [inline packages/web/.env.example:12]: 👍

File: packages/web/pages/checkout.tsx [hunk 2]
```diff
@@ -32,9 +39,12 @@ export default function Checkout() {
   const {
     value: { mode },
   } = useDeliveryOptions();
+  const [paymentReady, setPaymentReady] = useState(false);
+  const { total } = useCart();
+  const router = useRouter();
   return (
     <Layout open>
-      <Box margin="medium">
+      <Box margin="medium" gap="large">
```
Comment (jfplataroti) [inline packages/web/pages/checkout.tsx:47]: I think `gap` and `pad` ended up being my favourites attributes from `Box` heheh.

File: packages/web/components/paymentForm.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    setDisabled(event.empty);
+    setError(event.error ? event.error.message : '');
+    onPaymentStatusChange(false);
+    if (event.complete) {
+      handleSubmit(event);
+    }
+  };
+
+  const handleSubmit = async (ev) => {
+    console.log('handleSubmit');
```
Comment (jfplataroti) [inline packages/web/components/paymentForm.tsx:41]: we should remove this console.log.

**Review by jfplataroti — APPROVED** [review]
Nice work! simple code easy to understand! 👍 

I couldn't fully test it locally because I don't have the stripe key, but everything else looks great!.

File: packages/web/pages/checkout.tsx (hunk 2 above)
Comment (djoyal) [reply packages/web/pages/checkout.tsx:47]: 💯  Yes

File: packages/web/components/paymentForm.tsx (hunk 3 above)
Comment (djoyal) [reply packages/web/components/paymentForm.tsx:41]: Good catch!

File: packages/web/lib/getStripe.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Stripe, loadStripe } from '@stripe/stripe-js';
+import { env } from 'process';
+
+let stripePromise: Promise<Stripe | null>;
+const getStripe = () => {
+  if (!stripePromise) {
+    stripePromise = loadStripe(process.env.STRIPE_KEY);
```
Comment (jfplataroti) [inline packages/web/lib/getStripe.ts:7]: @djoyal to be able to make it work I had to use `NEXT_PUBLIC_STRIPE_KEY`, was it working for you with just `STRIPE_KEY`? 

I think only pass to the client-side the ones with the `NEXT_PUBLIC_` prefix.

Comment (djoyal) [reply packages/web/lib/getStripe.ts:7]: 🤔 It did seem fine, but maybe it was cached (I initially had it hardcoded, I can't remember if I restarted the NextJS environment after changing it) when I tested it. I'll update the key. Good catch!

**Pull request merged**